site/app/controllers/pdf/PDFController.php

```diff
@@ -174,12 +167,19 @@ public function savePDFAnnotation($gradeable_id, $target_dir) {
         if (!is_dir($annotation_version_path) && !FileUtils::createDir($annotation_version_path)) {
             return $this->core->getOutput()->renderJsonFail('Creating annotation version folder failed.');
         }
-
-        if (strpos($annotation_info['file_name'], '..') !== false) {
-            return false;
+        
+        $partial_path = substr_replace($annotation_info["file_path"], "", 0, strlen($annotation_version_path) + 1);
+        
+        $annotation_layer_decoded = json_decode($_POST['annotation_layer'], true);
+        
+        for ($i = 0; $i < count($annotation_layer_decoded); $i++) {
+            $annotation_layer_decoded[$i]["file_path"] = $partial_path;
+            $annotation_layer_decoded[$i]["grader_id"] = $grader_id;
         }
```

Comment:
as I've said before, remove this. The file_path and grader_id information should be stored at the top level of the json file with annotations then stored underneath an `annotations` key. We should not be modifying the annotations directly so as to not worry about interop issues with pdf-annotate.js.

Reply:
Namely, this structure:
```
{
  "grader_id": "instructor",
  "file_path": "summary 21 1 page.pdf",
  "annotations: [ ... ]
}
```

You'll have to modify where the annotation layers are being read into memory for display either in the PHP or JS to avoid the error you posted above in an unrelated comment.